src/Illuminate/Foundation/Testing/DatabaseTransactions.php

```diff
@@ -4,12 +4,42 @@
 
 trait DatabaseTransactions
 {
+    /**
+     * This property will store references to the connections currently affected by the transaction. They are stored
+     * as a key => value array of connection_name => ConnectionObject.
+     *
+     * @var array
+     */
+    protected $transactingConnections = [];
+
+    /**
+     * Get the names of the database connections that should be included in the transaction. The default value of empty
+     * string will cause only the default database connection to be included in the transaction.
+     *
+     * @return array
+     */
+    protected function connectionsToTransact()
+    {
+        return [''];
+    }
+
     public function beginDatabaseTransaction()
     {
-        $this->app->make('db')->beginTransaction();
+        $db = $this->app->make('db');
+
+        foreach ($this->connectionsToTransact() as $connectionName) {
+            $connection = $db->connection($connectionName);
```

Comment:
Do you want it to be 

```
$this->transactingConnections[$connectionName] = $db->connection($connectionName);
$this->transactingConnections[$connectionName]->beginTransaction();
```

If that's what you want I can do that, but that seems worse to me.

And it can't be done in a one liner since beginTransaction returns void and I need that assignment to the currently transacting array.